specs/merge/beacon-chain.md

```diff
@@ -59,148 +60,138 @@ We define the following Python custom types for type hinting and readability:
-
-The following methods are added to the `ExecutionEngine` protocol for use in the state transition:
-
-#### `new_block`
+## Helper functions
 
-Verifies the given `execution_payload` with respect to execution state transition, and persists changes if valid.
+### Predicates
 
-The body of this function is implementation dependent.
-The Consensus API may be used to implement this with an external execution engine.
+#### `is_merge_complete`
 
 ```python
-def new_block(self: ExecutionEngine, execution_payload: ExecutionPayload) -> bool:
-    """
-    Returns True if the ``execution_payload`` was verified and processed successfully, False otherwise. 
-    """
-    ...
+def is_merge_complete(state: BeaconState) -> bool:
+    return state.latest_execution_payload_header != ExecutionPayloadHeader()
 ```
 
-## Helper functions
+#### `is_merge_block`
 
-### Misc
+```python
+def is_merge_block(state: BeaconState, body: BeaconBlockBody) -> bool:
+    return not is_merge_complete(state) and body.execution_payload != ExecutionPayload()
+```
 
 #### `is_execution_enabled`
 
 ```python
-def is_execution_enabled(state: BeaconState, block: BeaconBlock) -> bool:
-    return is_transition_completed(state) or is_transition_block(state, block)
+def is_execution_enabled(state: BeaconState, body: BeaconBlockBody) -> bool:
+    return is_merge_block(state, body) or is_merge_complete(state)
 ```
 
-#### `is_transition_completed`
+### Misc
 
-```python
-def is_transition_completed(state: BeaconState) -> bool:
-    return state.latest_execution_payload_header != ExecutionPayloadHeader()
-```
+#### `compute_timestamp_at_slot`
 
-#### `is_transition_block`
+*Note*: This function is unsafe with respect to overflows and underflows.
 
 ```python
-def is_transition_block(state: BeaconState, block: BeaconBlock) -> bool:
-    return not is_transition_completed(state) and block.body.execution_payload != ExecutionPayload()
+def compute_timestamp_at_slot(state: BeaconState, slot: Slot) -> uint64:
+    slots_since_genesis = slot - GENESIS_SLOT
+    return uint64(state.genesis_time + slots_since_genesis * SECONDS_PER_SLOT)
 ```
 
-#### `compute_time_at_slot`
+## Beacon chain state transition function
 
-*Note*: This function is unsafe with respect to overflows and underflows.
+### Execution engine
+
+The implementation-dependent `ExecutionEngine` protocol encapsulates the execution sub-system logic via:
+
+* a state object `self.execution_state` of type `ExecutionState`
+* a state transition function `self.on_payload` which mutates `self.execution_state`
+
+#### `on_payload`
 
 ```python
-def compute_time_at_slot(state: BeaconState, slot: Slot) -> uint64:
-    slots_since_genesis = slot - GENESIS_SLOT
-    return uint64(state.genesis_time + slots_since_genesis * SECONDS_PER_SLOT)
+def on_payload(self: ExecutionEngine, execution_payload: ExecutionPayload) -> bool:
+    """
+    Returns ``True`` iff ``execution_payload`` is valid with respect to ``self.execution_state``.
```

Comment:
```suggestion
    Returns ``True`` if ``execution_payload`` is valid with respect to ``self.execution_state``.
```

Reply:
This is not a typo, see https://en.wikipedia.org/wiki/If_and_only_if

Reply:
Wow, I didn't know that! this is cool 👍